aten/src/ATen/native/Distance.cpp

```diff
@@ -22,6 +23,32 @@ Tensor pdist(const Tensor& self, const double p) {
   return at::_pdist_forward(self.contiguous(), p);
 }
 
+Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
+    AT_CHECK(x1.dim() == 2, "cdist only supports 2D tensors, X1 got: ", x1.dim(), "D");
+    auto device1 = x1.type().device_type();
+    AT_CHECK(device1 == kCPU || device1 == kCUDA, "cdist only supports CPU and CUDA devices, X1 got: ", device);
+    AT_CHECK(x2.dim() == 2, "cdist only supports 2D tensors, X2 got: ", x2.dim(), "D");
+    auto device2 = x2.type().device_type();
+    AT_CHECK(device2 == kCPU || device2 == kCUDA, "cdist only supports CPU and CUDA devices, X2 got: ", device);
+    AT_CHECK(p >= 0, "cdist only supports non-negative p values");
+    AT_CHECK(device1 == device2, "X1 and X2 must have the same device type. X1: ", device1, " X2: ", device2);
```

Comment:
I'm actually not sure what other ops check here (though this is probably okay).   Can you look at some other examples and report back?  For example, what happens if one op of `pdist` is CPU and one is CUDA?  Or one is `cuda:0` and the other is `cuda:1`?

Reply:
Checked these testcases:

1) CPU and CUDA devices
>>> import torch;
>>> torch.cuda.is_available()
True
>>> x = torch.randn((3, 5), device='cpu')
>>> y = torch.randn((4, 5), device='cuda')
>>> torch.cdist(x, y, 2)
Segmentation fault (core dumped)

2) Different CUDA devices
>>> import torch;
>>> print(torch.cuda.current_device())
0
>>> x = torch.randn((3, 5), device='cuda')
>>> x
tensor([[ 1.4912,  0.0627,  0.2258, -0.0317, -1.5003],
        [-0.9519, -1.0205,  1.0646,  0.0418, -0.6407],
        [ 0.4870,  1.4298, -0.3464, -1.5242, -1.2492]], device='cuda:0')
>>> torch.cuda.set_device(1)
>>> y = torch.randn((4, 5), device='cuda')
>>> y
tensor([[ 1.2308, -1.4090,  0.8967, -1.4265, -0.0454],
        [-0.5917, -0.6217,  1.0569, -1.4225, -1.2125],
        [ 1.0328, -0.6643, -0.2227, -1.4387,  1.5642],
        [-2.7521,  0.3133,  0.3392, -0.9948,  0.0224]], device='cuda:1')
>>> torch.cdist(x, y, 1)
THCudaCheck FAIL file=/data/users/ifedan/pytorch/aten/src/THC/THCReduceAll.cuh line=317 error=77 : an illegal memory access was encountered

So I added correspondent check to validate that CUDA devices have same index